### PR TITLE
Filter out extra lines from probe output

### LINKED ISSUE #21616: Toolchain probing cannot cope with warnings from java
### Expected Behavior

Gradle should be able to handle "java -version" output that contains warnings:

```console
$ java -version
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
openjdk version "17.0.4" 2022-07-19 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS, mixed mode, sharing)
```

### Current Behavior

Gradle (7.5.1) `javaToolchains` task fails with:

```
(14:47:50)  + Options
(14:47:50)      | Auto-detection:     Enabled
(14:47:50)      | Auto-download:      Enabled
(14:47:50) 
(14:47:50)  + Red Hat, Inc. JDK 17.0.4+8-LTS
(14:47:50)      | Location:           /usr/lib/jvm/java-17-openjdk-17.0.4.0.8-2.el8_6.x86_64
(14:47:50)      | Language Version:   17
(14:47:50)      | Vendor:             Red Hat, Inc.
(14:47:50)      | Architecture:       amd64
(14:47:50)      | Is JDK:             true
(14:47:50)      | Detected by:        Common Linux Locations
(14:47:50) 
(14:47:50)  + Invalid toolchains
(14:47:50)      + /usr/lib/jvm/java-11-openjdk-11.0.16.0.8-1.el8_6.x86_64
(14:47:50)        | Error:              Unexpected command output: 
(14:47:50) [0.015s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
(14:47:50) [0.015s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
(14:47:50) [0.015s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
(14:47:50) /usr/lib/jvm/java-11-openjdk-11.0.16.0.8-1.el8_6.x86_64
(14:47:50) 11.0.16
(14:47:50) Red Hat, Inc.
(14:47:50) amd64
(14:47:50) OpenJDK 64-Bit Server VM
(14:47:50) 11.0.16+8-LTS
(14:47:50) OpenJDK Runtime Environment
(14:47:50) 11.0.16+8-LTS
```

### Context

This happens when we run Podman containers on OpenShift - Podman runs as non-root. And Gradle inside of this.
So this is a two-layer deep container context (on top of core OS).

Lots of interesting OCP/Podman/Kernel magic is in play that I cannot possibly describe fully.
But I suspect that the JDK warns about the fact that it runs in a container - or some aspect of this, which it is surprised about.

### Steps to Reproduce 

It is not easy to reproduce (as per Context). But I will be super happy to help test a fix!

I can provide you with this output, which I hope is enough for you to accept the issue.

The output is from running `java -version` (showing what is stdout and what is stderr):

```
STEP 2/4: RUN java -version
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.056s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
openjdk version "17.0.4" 2022-07-19 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS, mixed mode, sharing)
--> c52c806705f
STEP 3/4: RUN java -version >/dev/null
openjdk version "17.0.4" 2022-07-19 LTS
OpenJDK Runtime Environment (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.4.0.8-2.el9_0) (build 17.0.4+8-LTS, mixed mode, sharing)
--> 955548f0dcc
STEP 4/4: RUN java -version 2>/dev/null
[0.004s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.004s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
[0.004s][warning][os,container] Duplicate cpuset controllers detected. Picking /sys/fs/cgroup/cpuset, skipping /sys/fs/cgroup/cpuset.
```

### Other hints

The only other (relevant) reference to this problem with java that I could find is: https://github.com/NixOS/nixpkgs/issues/163164#issuecomment-1122582269

Without knowing all the intricacies of parsing output from all the possible variants and versions of java, I would suggest using only the stderr output from 'java -version'. But that may of course break other stuff.
Or maybe strip out lines beginning with [ if that is a standard for warnings used in all java variants/versions.

### PR DESCRIPTION
**Pull request by jbartok**
Fixes #21616

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test this

File: subprojects/jvm-services/src/main/java/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetector.java [hunk 1]
```diff
@@ -131,7 +132,10 @@ private static File javaExecutable(File jdkPath) {
     }
 
     private JvmInstallationMetadata parseExecOutput(File jdkPath, String probeResult) {
-        String[] split = probeResult.split(System.getProperty("line.separator"));
+        String[] split = Arrays.stream(probeResult.split(System.getProperty("line.separator")))
+                .filter(l -> l.startsWith(MetadataProbe.MARKER_PREFIX))
```
Comment (eskatos) [inline subprojects/jvm-services/src/main/java/org/gradle/internal/jvm/inspection/DefaultJvmMetadataDetector.java:136]: :lipstick: nitpick so it reads the same way as the next statement
```suggestion
                .filter(line -> line.startsWith(MetadataProbe.MARKER_PREFIX))
```

**Review by eskatos — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test this

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**